src/lib/operators/get_table.cpp

```diff
@@ -274,4 +311,86 @@ std::shared_ptr<const Table> GetTable::_on_execute() {
+  if (_prunable_subquery_scans.empty()) {
+    return {};
+  }
+
+  // Create a dummy PredicateNode for each predicate containing a subquery that has already been executed. We do not use
+  // the original predicate to ignore all other nodes between the StoredTableNode and the PredicateNodes. Since the
+  // ChunkPruningRule already took care to add only predicates that are safe to prune with, we can act as if there were
+  // no other LQP nodes.
+  auto prunable_predicate_nodes = std::vector<std::shared_ptr<PredicateNode>>{};
+  prunable_predicate_nodes.reserve(_prunable_subquery_scans.size());
+
+  // Create a dummy StoredTableNode from the table to retrieve. `compute_chunk_exclude_list` modifies the node's
+  // statistics and we want to avoid that. We cannot use `deep_copy()` here since it would complain that the referenced
+  // prunable PredicateNodes are not part of the LQP.
+  const auto& stored_table_node = static_cast<const StoredTableNode&>(*lqp_node);
+  const auto dummy_stored_table_node = StoredTableNode::make(_name);
+
+  for (const auto& op : prunable_subquery_predicates()) {
+    const auto& table_scan = static_cast<const TableScan&>(*op);
+    const auto& operator_predicate_arguments = table_scan.predicate()->arguments;
+    const auto& predicate_node = static_cast<const PredicateNode&>(*table_scan.lqp_node);
+    const auto adjusted_predicate = predicate_node.predicate()->deep_copy();
+    auto& arguments = adjusted_predicate->arguments;
+    const auto argument_count = adjusted_predicate->arguments.size();
+
+    // Adjust predicates with the dummy StoredTableNode and the subquery result, if available.
+    for (auto expression_idx = size_t{0}; expression_idx < argument_count; ++expression_idx) {
+      auto& argument = arguments[expression_idx];
+      // Replace any column with the respective column from our dummy StoredTableNode.
+      if (const auto lqp_column = std::dynamic_pointer_cast<LQPColumnExpression>(argument)) {
+        Assert(*lqp_column->original_node.lock() == stored_table_node,
+               "Predicate is performed on wrong StoredTableNode.");
+        argument = lqp_column_(dummy_stored_table_node, lqp_column->original_column_id);
+        continue;
+      }
+
+      // Check if expression is an uncorrelated subquery.
+      if (argument->type != ExpressionType::LQPSubquery) {
+        continue;
+      }
+      Assert(operator_predicate_arguments[expression_idx]->type == ExpressionType::PQPSubquery,
+             "Cannot resolve PQPSubqueryExpression.");
+      const auto& subquery = static_cast<PQPSubqueryExpression&>(*operator_predicate_arguments[expression_idx]);
+      if (subquery.is_correlated()) {
+        continue;
+      }
+
+      // Ignore the subquery if it has not been executed yet. A reason might be that scheduling the subquery before the
+      // GetTable operator would create a cycle. For instance, this can happen for a query like this:
```

Comment:
When do we actually deduplicate the GetTables? Is this a relevant case here? If so, should we (read: you) invest time in avoiding the deduplication?

Reply:
That's a good question, and we actually have two (connected) places. In the LQPTranslator, equivalent (same pruned columns and chunks) GetTable operators are deduplicated. The StoredTableColumnAlignmentRule tries to combine multiple StoredTableNodes (translated: GetTables) with dfferently pruned columns to avoid duplication. For now, prunable subquery predicates are part of hash/equals (so duplication whenever these predicates differ).

Reply:
(note that GetTables with different pruned columns were duplicated before, too)

Reply:
> For now, prunable subquery predicates are part of hash/equals (so duplication whenever these predicates differ).

Can we ever create such cycles then?

Reply:
Ah, now I get what you mean. Yes, the GetTable operator in the example would possibly be split into two operators

Reply:
Yeah, would make it more obvious why we check it here and why it might later be relevant.

Let's discuss in person.